lib/chef/data_collector/resource_report.rb

```diff
@@ -83,13 +85,38 @@ def to_hash
           hash["recipe_name"]      = new_resource.recipe_name
         end
 
-        hash["conditional"]   = conditional.to_text if status == "skipped"
+        hash["conditional"] = conditional.to_text if status == "skipped"
         hash["error_message"] = exception.message unless exception.nil?
 
         hash
       end
       alias :to_h :to_hash
       alias :for_json :to_hash
+
+      # We should be able to call the identity of a resource safely, but there
+      # is an edge case where resources that have a lazy property that is both
+      # the name_property and the identity property, it will thow a validation
+      # exception causing the chef-client run to fail. We are not fixing this
+      # case since Chef is actually doing the right thing but we are making the
+      # ResourceReporter smarter so that it detects the failure and sends dummy
```

Comment:
Rather than calling this "dummy data", how about we change that last part of the sentence to: "sends a message to the data collector containing a static resource identity since we were unable to generate a proper one."